src/Altinn.Platform/Altinn.Platform.Authorization/IntegrationTests/PolicyInformationRepositoryTest.cs

```diff
@@ -0,0 +1,326 @@
+
+            // Assert
+            Assert.Null(application);
+        }
+
+        /// <summary>
+        /// Test case: Get from cosmos an application where app is null
+        /// Expected: GetInstance throws ArgumentNullException
+        /// </summary>
+        [Fact]
+        public async Task GetApplication_TC03()
+        {
+            // Arrange
+            if (!databasePopulatedApplications)
+            {
+                await PopulateCosmosDbAsyncWithApplication();
+            }
+
+            // Act & Assert
+            await Assert.ThrowsAsync<ArgumentNullException>(() => _pir.GetApplication(null, ORG));
+        }
+
+        /// <summary>
+        /// Test case: Get from cosmos an application where org is null
+        /// Expected: GetInstance throws ArgumentNullException
+        /// </summary>
+        [Fact]
+        public async Task GetApplication_TC04()
+        {
+            // Arrange
+            if (!databasePopulatedApplications)
+            {
+                await PopulateCosmosDbAsyncWithApplication();
+            }
+
+            // Act & Assert
+            await Assert.ThrowsAsync<ArgumentNullException>(() => _pir.GetApplication(APP, null));
+        }
+
+        /// <summary>
+        /// Test case: Get from cosmos an exsisting application 
+        /// Expected: GetApplication returns application with the same app and org as sent in 
+        /// </summary>
+        [Fact]
+        public async Task GetApplication_TC05()
+        {
+            // Arrange
+            if (!databasePopulatedApplications)
+            {
+                await PopulateCosmosDbAsyncWithApplication();
+            }
+
+            // Act
+            Application application = await _pir.GetApplication(APP, ORG);
+
+            // Assert
+            Assert.Equal("tdd/cat", application.Id);
+            Assert.Equal(ORG, application.Org);
+        }
+
+        private async Task PopulateCosmosDbAsyncWithInstance()
+        {
+            Uri uri = UriFactory.CreateDocumentCollectionUri(databaseId, instanceCollectionId);
+            Stream dataStream = File.OpenRead("Data/Instances/50013976/f3fc6233-1631-429d-8405-e1678f88dbd7.json");
+            dataStream.Position = 0;
+
+            string json = "";
+
+            using (StreamReader sr = new StreamReader(dataStream))
+            {
+                json = sr.ReadToEnd();
+            }
+
+            JObject jObject = JObject.Parse(json);
+
+            try
+            {
+                await _client.CreateDocumentAsync(uri, jObject);
+                databasePopulatedInstances = true;
+            }
+            catch (Exception ex)
+            {
+                databasePopulatedInstances = false;
+            }
+        }
+
+        private async Task PopulateCosmosDbAsyncWithApplication()
+        {
+            Uri uri = UriFactory.CreateDocumentCollectionUri(databaseId, "applications");
```

Comment:
should we for the integration tests ensure that the DB exists?? Will always be the case in the clusters, but not nessesarily for the pc running these tests.

Reply:
I think creating a database or collection if not exists must be part of automated dev environment setup rather than part of each platform component